python/mpt/src/mpt/state_diff.py

```diff
@@ -0,0 +1,242 @@
+
+        Returns
+        -------
+        StateDiff
+            A StateDiff object containing:
+            - 'account_diffs': Dict[Address, AccountDiff] - Accounts that were added, modified or deleted with their storage changes
+        """
+        return compute_state_diff(pre_state, post_state)
+
+
+def compute_state_diff(pre_state: State, post_state: State) -> StateDiff:
+    """
+    Compute the difference between pre_state and post_state.
+
+    Parameters
+    ----------
+    pre_state : State
+        The state before executing a block
+    post_state : State
+        The state after executing a block
+
+    Returns
+    -------
+    StateDiff
+        A StateDiff object containing:
+        - 'account_diffs': Dict[Address, AccountDiff] - Accounts that were added, modified or deleted with their storage changes
+    """
+
+    diff = StateDiff()
+
+    pre_addresses = set(pre_state._main_trie._data.keys())
+    post_addresses = set(post_state._main_trie._data.keys())
+
+    logger.debug(
+        f"Computing state diff: {len(pre_addresses)} pre addresses, {len(post_addresses)} post addresses"
+    )
+
+    # Find deleted accounts
+    deletions = pre_addresses - post_addresses
+    if deletions:
+        logger.error(
+            f"Warning: Since EIP-6780, we should not see any accounts marked for deletion at the level of the block - found {len(deletions)}"
+        )
+        for address in deletions:
+            if get_account(pre_state, address) != EMPTY_ACCOUNT:
+                logger.error(
+                    f"Invariant broken - Address marked for deletion (not in post state): 0x{address.hex()}"
+                )
+            else:
+                logger.debug(
+                    f"✅ Address marked for deletion is an empty account: 0x{address.hex()}"
+                )
+            # Process deleted account
+            account_diff = compute_account_diff(pre_state, post_state, address)
+            if account_diff is not None:
+                logger.debug(f"Adding deletion diff for address: 0x{address.hex()}")
+                diff.account_diffs[address] = account_diff
+
+    # Find added or modified accounts
+    modified_count = 0
+    for address in post_addresses:
+        # Process added or modified account
+        account_diff = compute_account_diff(pre_state, post_state, address)
+        if account_diff is not None:
+            modified_count += 1
+            if address not in pre_addresses:
+                logger.debug(f"Account added (not in pre state): 0x{address.hex()}")
+            else:
+                logger.debug(f"Account modified: 0x{address.hex()}")
+            diff.account_diffs[address] = account_diff
+
+    logger.debug(f"Found {modified_count} accounts added or modified")
+    logger.debug(f"Total account diffs: {len(diff.account_diffs)}")
+
+    return diff
+
+
+def compute_account_diff(
+    pre_state: State,
+    post_state: State,
+    address: Address,
+) -> Optional[AccountDiff]:
```

Comment:
get the account and not the state as arg

Reply:
then we need the account & the post_state.storage_tries[address] in the args